client/state/data-layer/wpcom/jetpack-onboarding/index.js

```diff
@@ -12,9 +12,67 @@ import { translate } from 'i18n-calypso';
 import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
 import { errorNotice } from 'state/notices/actions';
 import { http } from 'state/data-layer/wpcom-http/actions';
-import { JETPACK_ONBOARDING_SETTINGS_SAVE } from 'state/action-types';
+import {
+	JETPACK_ONBOARDING_SETTINGS_REQUEST,
+	JETPACK_ONBOARDING_SETTINGS_SAVE,
+} from 'state/action-types';
+import { getUnconnectedSite } from 'state/selectors';
 import { updateJetpackOnboardingSettings } from 'state/jetpack-onboarding/actions';
 
+export const announceRequestFailure = ( { dispatch } ) =>
+	dispatch(
+		errorNotice( translate( 'Could not fetch settings from site. Please try again later.' ) )
+	);
+
+const receiveJetpackOnboardingSettings = (
+	{ dispatch },
+	action,
+	{ data: { onboarding: settings } }
+) => {
+	const { siteId } = action;
+
+	if ( ! settings ) {
+		return announceRequestFailure( { dispatch }, action );
+	}
+
+	dispatch( updateJetpackOnboardingSettings( siteId, settings ) );
+};
+
+/**
+ * Dispatches a request to fetch settings for a given site
+ *
+ * @param   {Object} action Redux action
```

Comment:
JSdoc is a little off here - we're missing the first `store` parameter.

Reply:
449f164